### DISCUSSION TITLE
[Plus] private 와 final의 차이

### DISCUSSION THREAD

**Discussion by yerim123456** [GDSC-SWU/2023-JAVA-Study#152] [Q&A] [answered]
상속 메서드를 private와 final로 지정했을 때의 차이가 무엇인가요?

**Comment by hyojeongchoi** [comment]
Item 8 42p "final이 아닌 클래스를 finalizer 공격으로부터 방어하려면 아무 일도 하지 않는 finalize 메서드를 만들고 final로 선언하자"

**Comment by nusuy** [comment, marked as answer]
메소드를 `private`으로 지정 -> 상속이 안 됨. (상속이 되지 않기 때문에 오버라이딩은 당연히 안 됨!)
메소드를 `final`으로 지정 -> 상속은 되지만, 오버라이딩이 안 됨.

**Reply by hyojeongchoi** [reply]
final이 아닌 클래스를 finalizer 공격으로부터 방어하려는거니까 저 final이 아닌 클래스를 상속받은 클래스 또한 저 공격으로부터 방어해야하므로 상속은 가능한 상태여야함
